wn/util.py

```diff
@@ -103,10 +105,14 @@ class ProgressBar(ProgressHandler):
     def update(self, n: int = 1) -> None:
         """Increment the count by *n* and print the reformatted bar."""
         self.kwargs['count'] += n  # type: ignore
-        s = self.format()
-        if self.file:
-            print('\r\033[K', end='', file=self.file)
-            print(s, end='', file=self.file)
+        self._refresh_quota -= n
+        draw_bar = self._refresh_quota <= 0
+        if draw_bar:
```

Comment:
Why not just the following?

```python
if self._refresh_quota <= 0:
```

I don't think we need to declare a new variable to hold a one-off comparison value.

Reply:
Sorry about that, it was kind of a leftover of something silly I have done previously. Addressed.